crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -81,14 +81,44 @@ impl ReportFormatter<PubGrubPackage, Range<Version>, UnavailableReason>
                     format!("there is no version of {package}{set}")
                 } else {
                     let complement = set.complement();
-                    let segments = complement.iter().count();
-                    // Simple case, there's a single range to report
-                    if segments == 1 {
+                    let segments: Vec<_> = complement.iter().collect();
+
+                    // A single segment
+                    if let [(lower, upper)] = segments.as_slice() {
+                        // If the bound is inclusive, and the version is _not_
+                        // available, change it to an exclusive bound.
+                        let lower = match lower {
```

Comment:
Suggestions welcome on the pattern here, but I'm going to open an issue to follow-up to make this reusable in the multi-segment case as well.